src/core/expr.rs

```diff
@@ -72,6 +136,106 @@ impl UnFn {
     }
 }
 
+impl From<UnFn> for FnAssume {
+    /// Returns the default [FnAssume] function assumptions for every [UnFn]
+    fn from(func: UnFn) -> FnAssume {
```

Comment:
Very cool! So this is where all the axioms for a function is specified. Some non-actionable thought: It is currently quite version to express some of the constraints. Maybe we can borrow a page from [egg's pattern language][egg-pl] and build something more concise in the future.

[egg-pl]: https://docs.rs/egg/latest/egg/#simple-example

Reply:
Yep! This is what I was thinking as well, adding a `From<String> for Expr` would make it much easier to do!

Reply:
Great! We can make an issue to maybe revisit this as our axiom set becomes larger.